npm/ux/scatterChart.tsx

```diff
@@ -202,46 +196,93 @@ export function ScatterChart(props: {
   }`;
 
   const yAxisTextPaddingFromTicks = 5;
-  const yAxisTextYPadding = 6;
+  const yAxisTextYPadding = 4;
+
+  function trySetAttribute(element: Element, attribute: string, value: string) {
+    if (element.getAttribute(attribute) !== value) {
+      element.setAttribute(attribute, value);
+    }
+  }
+
+  function hideHoverTooltip() {
+    const tooltip = chart?.querySelector("#tooltip-hover");
+
+    if (tooltip) {
+      if (tooltip.getAttribute("clicked") === "false") {
+        tooltip.setAttribute("visibility", "hidden");
+      }
+    }
+  }
+
+  const scatterChartContainerRef = useRef<SVGSVGElement | null>(null);
+  let chart: Element | null = null;
+
+  function updateCoordinates() {
+    if (!chart) {
+      return;
+    }
+
+    chart.querySelectorAll("[x-data-value]").forEach((element) => {
+      const value = Number(element.getAttribute("x-data-value"));
+      const x = coordinateToSvgLogarithmic(value, rangeX, plotAreaWidth);
+      const padding = element.getAttribute("x-data-padding");
+      const value_with_padding = (x + Number(padding)).toString();
+      trySetAttribute(element, "x", value_with_padding);
+      trySetAttribute(element, "x1", value_with_padding);
+      trySetAttribute(element, "x2", value_with_padding);
+      trySetAttribute(element, "cx", value_with_padding);
+    });
+
+    chart.querySelectorAll("[y-data-value]").forEach((element) => {
+      const value = Number(element.getAttribute("y-data-value"));
+      const y = -coordinateToSvgLogarithmic(value, rangeY, plotAreaHeight);
+      const padding = element.getAttribute("y-data-padding");
+      const value_with_padding = (y + Number(padding)).toString();
+      trySetAttribute(element, "y", value_with_padding);
+      trySetAttribute(element, "y1", value_with_padding);
+      trySetAttribute(element, "y2", value_with_padding);
+      trySetAttribute(element, "cy", value_with_padding);
+    });
+  }
+
+  useEffect(() => {
+    chart = scatterChartContainerRef.current as Element | null;
+    updateCoordinates();
+  });
 
   return (
     <div style="display: flex; flex-wrap: wrap; margin-top: 8px;">
       <div class="chart-container">
         <svg
+          class="qs-scatterChart-svg"
```

Comment:
I don't see this class defined in the CSS. Is it used for any styling or just a marker?